packages/contracts/src/Game.sol

```diff
@@ -295,6 +298,13 @@ contract Game {
         gdata.lastBlockNum = block.number;
     }
 
+    // Checks that the game exists, that the requested step is compatible with the game state, and
+    // that the next player is the message sender.
+    modifier step(uint256 gameID, GameStep requestedStep) {
+        _checkLegitStep(gameId,requestedStep);
+        _;
+    }
+
```

Comment:
This is a good idea ... but does it compile? You left `_;` in checkLegitStep! It needs to be split into two functions (one for before, and one for after). Probably a good idea to extract the gameData pointer in the step logic and pass it to both functions.